### PR TITLE
🌱 bubble-dev/start-preset: support passing template name to the pkg task

### PR DESCRIPTION
**Pull request by moriaam**
This pr takes care of:
1. Being able to pass the template name in advance
2. Not being prompted for a package name (it was already working automatically, but we were still prompted)

### CODE REVIEW

File: packages/bubble-dev/start-preset/src/pkg.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export type TReplacers = {
   },
 }
 
-export const Pkg = (replacers?: TReplacers) => (packagePath: string, shouldAddMissingWorkspace: boolean = true) =>
+export const Pkg = (replacers?: TReplacers) => (packagePath: string, shouldAddMissingWorkspace: boolean = true, templateName: (string) = '') =>
```
Comment (irminakacz) [inline packages/bubble-dev/start-preset/src/pkg.ts:27]: Why `string` is in parenthesis here?

File: packages/bubble-dev/start-preset/src/pkg.ts [hunk 2]
```diff
@@ -62,44 +62,51 @@ export const Pkg = (replacers?: TReplacers) => (packagePath: string, shouldAddMi
...
+            onCancel: () => {
+              throw 'canceled'
+            },
+          }
+        ) as { type: string }
+
+        type = result.type
+      }
+
+      const packageNameResult = await prompts(
```
Comment (irminakacz) [inline packages/bubble-dev/start-preset/src/pkg.ts:93]: Why this change is needed? It doesn't seem you're using `packageNameResult` anywhere else, besides getting the `name` out of it lower down in the code?

File: packages/bubble-dev/start-preset/src/pkg.ts (hunk 1 above)
Comment (moriaam) [reply packages/bubble-dev/start-preset/src/pkg.ts:27]: old remainders from allowing it to be null 🙈
removed

File: packages/bubble-dev/start-preset/src/pkg.ts (hunk 2 above)
Comment (moriaam) [reply packages/bubble-dev/start-preset/src/pkg.ts:93]: I just made it the same as in type both for keeping consistency and to allow me to define the name variable earlier (as type was defined)

File: packages/bubble-dev/start-preset/src/pkg.ts [hunk 3]
```diff
@@ -62,44 +62,51 @@ export const Pkg = (replacers?: TReplacers) => (packagePath: string, shouldAddMi
...
     let type = null
+    let name = null
 
-    if (templateNames.length === 1) {
-      type = templateNames[0]
+    if (Boolean(templateName) && templateNames.includes(templateName)) {
+      type = templateName
     } else {
-      const result = await prompts(
+      if (templateNames.length === 1) {
```
Comment (DonGideon) [inline packages/bubble-dev/start-preset/src/pkg.ts:70]: can we add some comet here? something for future us to know in which stage of the script are we?

**Comment by irminakacz** [comment]
Have you manually tested both cases in bubble-next?

**Comment by moriaam** [comment]
> Have you manually tested both cases in bubble-next?

I tested all cases on bubble-ui
the pkg task is identical for bubble-ui and bubble-next

File: packages/bubble-dev/start-preset/src/pkg.ts (hunk 3 above)
Comment (moriaam) [reply packages/bubble-dev/start-preset/src/pkg.ts:70]: not sure I understand, wdym?

**Comment by irminakacz** [comment]
> I tested all cases on bubble-ui
Ah, yes - I meant bubble-ui 😅 That's great! Approved ✅

**Review by irminakacz — APPROVED** [review]

File: packages/bubble-dev/start-preset/src/pkg.ts (hunk 3 above)
Comment (DonGideon) [reply packages/bubble-dev/start-preset/src/pkg.ts:70]: i had in mind something to explain what does it mean that `templateNames.length === 1` just because it always feels uncomfortable for me to see something like `type = templateNames[0]`
or you do not feel there is a reason for it?

Comment (moriaam) [reply packages/bubble-dev/start-preset/src/pkg.ts]: oh I understand now! I strive for the code to be readable on its own when possible so I made it into a function
is it clearer? https://github.com/bubble-dev/_/pull/463/commits/3714f2d34682e0e5dd5dd228cff28f07fa52bbca#diff-1c161cfb97b16f26166bb64f3112736cba3eee7b1682442e0348df90166f8cadL70

**Review by DonGideon — APPROVED** [review]
